mindsdb/integrations/handlers/writer_handler/evaluator.py

```diff
@@ -0,0 +1,13 @@
+from scipy.spatial import distance
+
+
+def calculate_cosine_similarity(
+    context_embeddings: list, retrieved_embeddings: list
+) -> float:
+    cosine_sim = 1 - distance.cosine(context_embeddings, retrieved_embeddings)
+
+    return cosine_sim
```

Comment:
I propose we use cosine similarity to check retrieved context against context in the test data set for a given question